modules/gapi/src/backends/common/serialization.cpp

```diff
@@ -292,25 +292,83 @@ I::IStream& operator >> (I::IStream& is, cv::gapi::wip::IStreamSource::Ptr &)
     return is;
 }
 
-I::OStream& operator<< (I::OStream& os, const cv::detail::VectorRef &)
+namespace
+{
+template<typename Ref, typename T, typename... Ts>
+struct putToStream;
+
+// FIXME: workaround to end the recursion
+template<typename Ref>
+struct putToStream<Ref, std::tuple<>>
+{
+    static void put(I::OStream&, const Ref &)
+    {}
+};
```

Comment:
If you've reached the end of your type list, you should throw an exception.

Reply:
Nice catch, added